docs/adr/adr-008-monitoring-and-alerting.md

```diff
@@ -0,0 +1,269 @@
+[Travel Advice Publisher]: https://github.com/alphagov/travel-advice-publisher
+
+## Decision
+
+We decided that the system should raise a critical alert
+[if and only if](https://en.wikipedia.org/wiki/If_and_only_if) the email alert
+system is broken. The majority of these alerts should contact engineers
+out-of-hours.
+
+Unpacking this, we consider the email alert system to be comprised of 3
+applications: Email Alert API,
+[Email Alert Frontend](https://github.com/alphagov/email-alert-frontend) and
+[Email Alert Service](https://github.com/alphagov/email-alert-service); this
+excludes the monitoring of whether GOV.UK publishing and Notify are working
+as they are independently monitored. The use of "if and only if"
+reflects a condition we've formed that the system will not raise critical
+alerts unless we're highly confident it is broken, whereas warnings are
+appropriate for other situations. We decided that for the email alert system
+to be "broken" it represents a failing that will lead to a user noticeable
+issue unless an engineer intervenes. Since these alerts should represent
+a broken system requiring intervention, it should be suitable to contact
+engineers out-of-hours for them as otherwise users will be experiencing issues.
+
+We decided that we will re-approach our alerting with this mindset and look to
+replace alerts that are vulnerable to reflecting a busy status, or which
+extend beyond the boundaries of the system.
+
+## Status
+
+Pending
+
+## Consequences
+
+We will be changing the approaching to monitoring and alerting across the email
+alert system of applications in order to meet this new alerting criteria.
+
+### Monitoring Email Alert API
+
+#### Sidekiq workers
+
+We will configure critical alerts, that contact engineers out-of-hours, if
+individual Sidekiq jobs fail multiple times for the following workers:
+
+- [DailyDigestInitiatorWorker][]
+- [DigestEmailGenerationWorker][]
+- [ProcessContentChangeWorker][]
+- [ProcessMessageWorker][]
+- [SendEmailWorker][]
+- [WeeklyDigestInitiatorWorker][]
+
+These workers will use the database to store the number of times we have
+attempted to perform an action. An alert will be raised if the quantity of
+failed attempts reaches a threshold (the suggestion is 5 - based on this being
+evidence of consistent failure). To ensure that these jobs actually run we
+will configure the [RecoverLostJobsWorker][] to identify any of these that
+need to run and attempt them, this protects from lost work problems such
+as [Sidekiq crashing][sidekiq-reliability].
+
+We expect [SendEmailWorker][] and [DigestEmailGenerationWorker][] jobs to run
+quickly (durations of up to a few seconds) and know that there can be high
+volumes of them to run (both can regularly reach volumes of > 100,000).
+Therefore, we are concerned about a scenario where these consistently fail
```

Comment:
```suggestion
Therefore, we are concerned about a scenario where these consistently fail,
```

(They're not failing because it could take a long...)